src/Sylius/Component/Taxonomy/Generator/TaxonSlugGenerator.php

```diff
@@ -52,6 +52,6 @@ public function generate($name, $parentId = null)
         $parent = $this->taxonRepository->find($parentId);
         Assert::notNull($parent, sprintf('There is no parent taxon with id %d.', $parentId));
 
-        return $parent->getSlug().self::SLUG_SEPARATOR.$taxonSlug;
+        return $parent->getTranslation($locale)->getSlug().self::SLUG_SEPARATOR.$taxonSlug;
```

Comment:
`TranslatableTrait`:
```php
public function getTranslation($locale = null)
{
    $locale = $locale ?: $this->currentLocale;
    if (null === $locale) {
        throw new \RuntimeException('No locale has been set and current locale is undefined.');
    }

    // ...
}
```

Current locale is always set by a listener running after an object is loaded from the ORM.